lib/puppet/provider/f5_sslcertificate/rest.rb

```diff
@@ -0,0 +1,41 @@
+require 'puppet/provider/f5'
+require 'json'
+
+Puppet::Type.type(:f5_sslcertificate).provide(:rest, parent: Puppet::Provider::F5) do
+
+  def create_message(basename, hash)
+    # Create the message by stripping :present.
+    new_hash            = hash.reject { |k, _| [:ensure,:name, :provider, Puppet::Type.metaparams].flatten.include?(k) }
+  #  new_hash[:name]     = basename
+
+    return new_hash
+  end
+
+  def message(object)
+    # Allows us to pass in resources and get all the attributes out
+    # in the form of a hash.
+    message = object.to_hash
+
+    # Map for conversion in the message.
+    map = {
```

Comment:
`map` isn't used?

Reply:
Not used now, but leave it there for possible future use

Reply:
I disagree.  Leaving dead unused code everywhere is not helpful.

Reply:
I have to agree with @alexjfisher here.  If you need to reference the mapping model for there are plenty of providers where you can copy that code at the time of need @ericzji.  Unused code in general should be removed.  Furthermore we should get you running your code through [rubocop](https://github.com/bbatsov/rubocop).

Reply:
Removed.